dlrover/python/master/hyperparams/simple_strategy_generator.py

```diff
@@ -81,68 +62,78 @@ def generate_opt_strategy(
-        n_embd = model_config["n_embd"]
-
-        # Calculate the memory required for intermediate activation
-        activation_memory_gb = (
-            (
-                34 * batch_size * block_size * n_embd
-                + 5 * batch_size * (block_size**2) * n_heads
+        if min_remain_memory > 2400:
+            # Update dataloader configuration version
+            updated_version = dataloader_config.version + 1
+            # Extract dataloader configuration values
+            batch_size = dataloader_config.batch_size
+
+            # Extract model configuration values
+            block_size = model_config["block_size"]
+            n_layer = model_config["n_layer"]
+            n_heads = model_config["n_heads"]
+            n_embd = model_config["n_embd"]
+
+            # Calculate the memory required for intermediate activation
+            activation_memory_mb = (
+                (
+                    34 * batch_size * block_size * n_embd
+                    + 5 * batch_size * (block_size**2) * n_heads
+                )
+                * n_layer
+                / (1024**2)
             )
-            * n_layer
-            / (1024 * 1024 * 1024)
-        )
-
-        try:
-            updated_batch_size = int(
-                batch_size
-                + batch_size * min_remain_memory / activation_memory_gb
+            try:
+                updated_batch_size = int(
+                    batch_size
+                    + batch_size * min_remain_memory / activation_memory_mb
+                )
+            except ZeroDivisionError:
+                updated_batch_size = batch_size
+
+            logger.info(f"updated_batch_size: {updated_batch_size}")
+            return DataLoaderConfig(
+                updated_version,
+                dataloader_config.dataloader_name,
+                updated_batch_size,
+                0,
+                0,
             )
-        except ZeroDivisionError:
-            updated_batch_size = batch_size
-
-        return DataLoaderConfig(
-            updated_version,
-            dataloader_config.dataloader_name,
-            updated_batch_size,
-            0,
-            0,
-        )
+        else:
+            return dataloader_config
 
     def _generate_optimizer_config(self):
         return OptimizerConfig(5, 6)
 
     def _extract_node_resource(self) -> Dict[str, List[List[Node]]]:
+        stats = self._stats_collector.get_runtime_stats()
         node_used_resources: Dict[str, List[List[Node]]] = {}
         node_used_resources[NodeType.WORKER] = []
-        simple_node = Node(node_type=NodeType.WORKER, node_id=0)
-        simple_node.used_resource.gpu_stats = gpu_stats
-        simple_node.paral_config.dataloader = mock_dataloader_config
-        simple_node.name = "simple_node"
-        node_used_resources[NodeType.WORKER].append([simple_node])
-        return node_used_resources
+        if len(stats) == 0:
+            logger.info("stats length is 0")
```

Comment:
`logger.info("There is no any training stats.")`

Reply:
Fixed in latest commit.